sdk/src/trace/tracer_provider.cc

```diff
@@ -14,7 +14,8 @@ opentelemetry::nostd::shared_ptr<opentelemetry::trace::Tracer> TracerProvider::G
     nostd::string_view library_name,
     nostd::string_view library_version) noexcept
 {
-  return opentelemetry::nostd::shared_ptr<opentelemetry::trace::Tracer>(tracer_);
+  // TODO: do we have to transform this at all, if we can keep it nostd::shared_ptr at source?
+  return tracer_;
```

Comment:
Basically I have a question why did we have to change the `tracer_` type to `std::shared_ptr` , that we eventually end up converting to `nostd::shared_ptr` .. Why don't we maintain this as original type, and then if we use STD library - there is no unnecessary type conversion back and forth.